drafts/samesite-cookies.adoc

```diff
@@ -1,134 +1,157 @@
 
-The `httpEndpoint` configuration also allows the use of the wildcard character (`*`).
+If the `None` value is used for `sameSiteCookie`, the `Secure` attribute is automatically added to the security cookies.
 
-* Wildcard Support:
-. As a standalone character:
+=== Setting the SameSite attribute for other cookies
+
+In the `server.xml`:
+
+[source,xml]
+----
+<httpEndpoint id="defaultHttpEndpoint"
+              httpPort="9080"
+              httpsPort="9443" >
+   <samesite lax="chocolateChip,oatmeal" strict="peanutButter" none="sugar"/>
+</httpEndpoint>
+----
+
+Each attribute of the `samesite` element takes a comma (`,`) deliminted list of cookie names.
 
-   <samesite lax="*"/>
+In the example, cookies with the following names would have the `SameSite` attribute defined as follows:
+- `chocolateChip` and `oatmeal` have a `SameSite` attribute of `Lax`
+- `peanutButter` have a `SameSite` attribute of `Strict`
+- `sugar` have a `SameSite` attribute of `None` and the `Secure` attribute is automatically added
 
-. At the end of one or more cookie names:
+The `httpEndpoint` configuration adds the `SameSite` attribute to a cookie only if it does not already exist.
 
-   <!--sugar would have the None SameSite attribute value as well as Secure attribute added -->
-   <!-- peanutButter would have the Strict SameSite attribute value -->
-   <!-- all other cookies would have the Lax SameSite attribute value -->
-   <samesite lax="*" strict="peanutButter" none="suga*"/>
+The `httpEndpoint` configuration gives the ability to add the `SameSite` attribute to cookies added with the link:{url-prefix}/docs/ref/javaee/8/#package=javax/servlet/http/package-frame.html&class=javax/servlet/http/HttpServletResponse.html#setHeader-java.lang.String-java.lang.String-[HttpServletResponse.setHeader] and link:{url-prefix}/docs/ref/javaee/8/#package=javax/servlet/http/package-frame.html&class=javax/servlet/http/HttpServletResponse.html#addHeader-java.lang.String-java.lang.String-[HttpServletResponse.addHeader] methods, as well as any cookies added using link:{url-prefix}/docs/ref/javaee/8/#package=javax/servlet/http/package-frame.html&class=javax/servlet/http/HttpServletResponse.html#addCookie-javax.servlet.http.Cookie-#addHeader-java.lang.String-java.lang.String-[HttpServletResponse.addCookie(Cookie cookie)].
 
-If the `httpSession` and/or `webAppSecurity` configurations have not been set then given the above examples the Session cookie `SameSite` attribute value would be `Lax` as well as the Security cookies `SameSite` attribute value would be `Lax`.
+The link:{url-prefix}/docs/ref/config/#httpEndpoint.html#samesite[`httpEndpoint` configuration] also supports the wildcard character (`*`).
 
-Any number of cookie names can contain the wildcard character at the end and for any given cookie name the most specific pattern will be matched.
+In the following example, all cookies have the SameSite attribute set to `Lax`:
```

Comment:
I'd like to be a bit more clear here, all cookies is all cookies added by the application and the server, care should be taken if using a wildcard.

Reply:
Will change "all cookies" to "all cookies added by the application and the server" - is that okay?